src/translate.py

```diff
@@ -3082,7 +3097,7 @@ def compute_has_custom_return(nodes: List[Node]) -> None:
         for n in nodes:
             block_info = n.block.block_info
             assert isinstance(block_info, BlockInfo)
-            if block_info.has_custom_return or block_info.has_function_call:
+            if block_info.has_custom_return or block_info.has_function_call_return:
```

Comment:
This change is wrong: function calls act as boundaries for the "has custom return" computation because they clobber registers, and they do so even if the function returns `void` (sorry for not commenting this code properly)

Reply:
I'll revert it.
Is splitting the `has_return` into `could_have_return` + `has_custom_return` in `translate_ast()` OK? I think that's separate, right?

Reply:
Yeah, that part looked fine to me